pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionaryMultiColumnGroupKeyGenerator.java

```diff
@@ -177,6 +305,29 @@ private int getGroupIdForKey(FixedIntArray keyList) {
     return groupId;
   }
 
+  /**
+   * Helper method to get or create a list of group-id for a list of group key.
+   *
+   * @param keysList Group keys, that is a list of list of objects to be grouped
+   * @return Group ids
+   */
+  private int[] getGroupIdsForKey(int[][] keysList) {
+    List<Integer> groupIds = new ArrayList<>();
+    getGroupIdsForKeyHelper(keysList, new int[keysList.length], 0, groupIds);
+    return groupIds.stream().mapToInt(i->i).toArray();
```

Comment:
Avoid using stream in query engine for performance concern.
To avoid boxing/unboxing, you may use fastutil `IntArrayList` and `toIntArray()` to get the primitive array.